src/test/java/de/tum/in/www1/artemis/TextAssessmentIntegrationTest.java

```diff
@@ -112,6 +119,70 @@ public void testPrepareSubmissionForAssessment() {
+
+        // Create Exercise to save blocks & submissions
+        TextExercise textExercise = textExerciseUtilService.createSampleTextExerciseWithSubmissions(course, new ArrayList<>(textBlocks), submissionCount, submissionSize);
+        textExercise.setMaxPoints(1000D);
+        exerciseRepository.save(textExercise);
+
+        // Create a cluster and set minimal distance to < Threshold = 1
+        int[] clusterSizes = { 2 };
+        List<TextCluster> clusters = textExerciseUtilService.addTextBlocksToCluster(textBlocks, clusterSizes, textExercise);
+        double[][] minimalDistanceMatrix = new double[numberOfBlocksTotally][numberOfBlocksTotally];
+        // Fill each row with an arbitrary fixed value < 1 to stimulate a simple case of 10 automatic feedback suggestions
+        for (double[] row : minimalDistanceMatrix) {
+            Arrays.fill(row, 0.1);
+        }
+        clusters.get(0).blocks(textBlocks.stream().toList()).distanceMatrix(minimalDistanceMatrix);
+        textClusterRepository.saveAll(clusters);
+
+        // save textBLocks
+        textBlockRepository.saveAll(textBlocks);
+
+        var listOfSubmissions = textSubmissionRepository.getTextSubmissionsWithTextBlocksByExerciseId(textExercise.getId());
+        TextSubmission firstSubmission = listOfSubmissions.get(0);
+        TextSubmission secondSubmission = listOfSubmissions.get(1);
+
+        // Create and set a sample result for submissions
+        Result firstResult = createSampleResultForSubmission(firstSubmission);
+        Result secondResult = createSampleResultForSubmission(secondSubmission);
+
+        // Create a manual feedback and attach first block to it as reference
+        Feedback feedback = new Feedback().credits(10.0).type(FeedbackType.MANUAL).detailText("gj");
+        feedback.setReference(textBlocks.stream().toList().get(0).getId());
+
+        // Set feedback to result
+        firstResult.setFeedbacks(Collections.singletonList(feedback));
+        resultRepo.save(firstResult);
+        feedback.setResult(firstResult);
+        feedbackRepository.save(feedback);
+
+        automaticTextFeedbackService.suggestFeedback(secondResult);
+
+        assertThat(secondResult).as("saved result found").isNotNull();
+        assertThat(secondResult.getFeedbacks().get(0).getReference()).isEqualTo(textBlocks.stream().toList().get(1).getId());
+        assertThat(secondResult.getFeedbacks().get(0).getSuggestedFeedbackOriginSubmissionReference()).isEqualTo(firstSubmission.getId());
+        assertThat(secondResult.getFeedbacks().get(0).getSuggestedFeedbackParticipationReference()).isEqualTo(firstSubmission.getParticipation().getId());
+        assertThat(secondResult.getFeedbacks().get(0).getSuggestedFeedbackReference()).isEqualTo(feedback.getReference());
+    }
+
+    public Result createSampleResultForSubmission(Submission submission) {
+        Result result = new Result();
```

Comment:
there is also method called `addResultToSubmission` in `DatabaseUtilService`. Since you wanted to return result, not sure if it is useful in here, just suggestion 😊

Reply:
hmm, that's close but it returns a submission instead of a Result and it also runs an extra DB query. I think its more clean this way,